src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java

```diff
@@ -1,31 +1,113 @@
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
 import org.mockito.junit.jupiter.MockitoExtension;
 
+import java.util.*;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.*;
+
+
 @ExtendWith(MockitoExtension.class)
 class PaymentServiceTest {
 
+    private PaymentService service;
+    private UserRepository userRepository;
+    private PaymentRepository paymentRepository;
+    private ValidationService validationService;
+
+    @Captor
+    ArgumentCaptor<Payment> paymentArgumentCaptor;
+
 
     @BeforeEach
     void setUp() {
+        userRepository = mock(UserRepository.class);
+        paymentRepository = mock(PaymentRepository.class);
+        validationService = mock(ValidationService.class);
+        service = new PaymentService(userRepository,paymentRepository,validationService);
 
     }
 
     @Test
     void createPayment() {
+        User user = new User(1,"User",Status.ACTIVE);
+
+        when(userRepository.findById(1)).thenReturn(Optional.of(user));
+        service.createPayment(user.getId(),200.00);
+
+        verify(validationService).validateUserId(any());
+        verify(validationService).validateAmount(anyDouble());
+        verify(validationService).validateUser(any());
+        verify(userRepository).findById(anyInt());
+
+        paymentArgumentCaptor = ArgumentCaptor.forClass(Payment.class);
+        verify(paymentRepository).save(paymentArgumentCaptor.capture());
+
+        Payment payment = paymentArgumentCaptor.getValue();
+
+        assertThat(user.getId()).isEqualTo(payment.getUserId());
+        assertThat(payment.getMessage()).isEqualTo("Payment from user " + user.getName());
+        assertThat(payment.getAmount()).isEqualTo(200);
+
+    }
+
+    @Test
+    void comparingExpectedAndActualPayment(){
+        Payment actual = new Payment(1,200.00,"User");
+        User user = new User(1,"User",Status.ACTIVE);
+
+        when(userRepository.findById(1)).thenReturn(Optional.of(user));
+
+        when(paymentRepository.save(any())).thenReturn(actual);
+        Payment expected = service.createPayment(user.getId(),200.00);
+        paymentArgumentCaptor = ArgumentCaptor.forClass(Payment.class);
+        verify(paymentRepository).save(paymentArgumentCaptor.capture());
+
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    @Test
+    void createPaymentThrowExceptionIfUserIsNotFound(){
+        when(userRepository.findById(1)).thenThrow(new NoSuchElementException());
+        Assertions.assertThrows(NoSuchElementException.class,()->service.createPayment(1,200.00));
 
     }
 
     @Test
     void editMessage() {
+        Payment payment = new Payment(1,200.00,"");
+
+        service.editPaymentMessage(payment.getPaymentId(),"Message");
```

Comment:
Here we have to test "editPaymentMessage" method from PaymentService; so, we need to check the test based on this method.
Please change the test.